### ISSUE TITLE
reset_active_counters resets times and thread counts before some of the counters are evaluated

### ISSUE THREAD

**Issue by pagrubel** [STEllAR-GROUP/hpx#1859]
Here is an example output of these counters when employed with reset = false, when evaluating the counters for every sample in the inncabs fft benchmark  (using the csv output format):
```
idle_rate,average_time,average_overhead,average_phase,average_phase_overhead,cumulative_time,cumulative_overhead,cumulative,phases
5789,4736,6513,2664,3669,9.69448e+09,1.33494e+10,2.04679e+06,3.63784e+06
5590,4678,5930,2632,3337,1.91592e+10,2.43473e+10,4.09345e+06,7.2744e+06
5517,4659,5735,2622,3231,2.86174e+10,3.52693e+10,6.14018e+06,1.09102e+07
```

Here is the output when evaluating the counters for every sample  with reset = true
```
idle_rate,average_time,average_overhead,average_phase,average_phase_overhead,cumulative_time,cumulative_overhead,cumulative,phases
5769,4759,0,2678,0,8.81228e+09,0,2,9
5330,4637,0,2346,0,6.64024e+09,0,2,9
5325,4634,0,2350,0,6.63114e+09,0,1,15
```

**Comment by pagrubel** [comment]
Multiple resets are done if counters use the same underlying data. Counters that are affected include
/threads/time/average   (resets executed_threads_ & tfunc_times_ ..... should reset exec_times_  )
/threads/time/average-overhead  ( same ^ should also reset exec_times_ )
/threads/time/cumulative  ( resets tfunc_times should reset exec_times_)
/threads/idle-rate ( resets tfunc_times  should also reset exec_times_)
/threads/count/cumulative  (resets executed_threads )

and this behavior is for the counters for phases as well
Also since the underlying values are reset and used when getting counts for individual worker-threads and for total counts, those will also be reset by one or the other before the other queries for them.

src/runtime/threads/detail/thread_pool.cpp
src/runtime/threads/threadmanager.cpp

like the util get_and_reset_value.hpp   would be nice if we could somehow use that for underlying values when we reset and keep a copy until all the counts are satisfied, or have a special way to do derived counts

**Comment by pagrubel** [comment]
I'm thinking if I add a counter for tfunc_times I can use it and the two others for time/cumulative which counts exec_times and the /count/cumulative to come up with the rest of the counts at post processing for now
The rest are derived counters

**Comment by hkaiser** [comment]
[20:47]	patg: hkaiser: to answer your question from earlier the counter that gave me negative values was /threads/time/average-overhead

**Comment by hkaiser** [comment]
Fixed by merging #1887
